packages/auth/src/hooks/useLogoutChannel.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useRef } from "react";
+
+const LOGOUT_MESSAGE = "LOGOUT";
+let singleChannel: BroadcastChannel;
+
+/**
+ * Create a single broadcast channel
+ * to emit logout message to all tabs
+ *
+ * @returns
+ */
+const getChannel = () => {
+  if (!singleChannel) {
+    singleChannel = new BroadcastChannel("logoutChannel");
+  }
+
+  return singleChannel;
+};
+
+const useLogoutChannel = (onLogout: () => void) => {
+  const channel = getChannel();
+  // Prevents channel from being closed/reopened
+  const isSubscribed = useRef(false);
+
+  useEffect(() => {
+    if (!isSubscribed.current) {
+      channel.onmessage = (event) => {
+        if (event.data === LOGOUT_MESSAGE) {
+          onLogout();
+        }
+      };
+    }
+
+    return () => {
+      if (isSubscribed.current) {
+        channel.close();
+        isSubscribed.current = false;
+      }
+    };
+    // We only want this to run on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const postLogoutMessage = useCallback(() => {
+    channel?.postMessage(LOGOUT_MESSAGE);
```

Comment:
Maybe we could log an error message if they try to post a logout message and the channel isn't ready?

Reply:
Added in [add error log when no channel on post](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9504/commits/f156c503ec8f686f8dcc69770ac9cf15f6ea1b73)